src/Grid/grid.jl

```diff
@@ -1450,3 +1450,44 @@ struct SurfaceOrientationInfo
     #flipped::Bool
     #shift_index::Int
 end
+# This one is the same as `SurfaceOrientationInfo` but used in interfaces. It is calculated for neighbor facet relative to current facet
+struct InterfaceOrientationInfo
+    flipped::Bool
+    shift_index::Int
+end
+# TODO: make this `SurfaceOrientationInfo`?
+"""
+    InterfaceOrientationInfo(grid::AbstractGrid, this_face::FaceIndex, neighbor_face::FaceIndex)
+
+Orientation information for 2D interfaces. Such an interface can be 
+possibly flipped (i.e. the defining vertex order is reverse to the 
+spanning vertex order) and the vertices can be rotated against each other.
+Take for example the faces
+```
+1---2 2---3
+| A | | B |
+4---3 1---4
+```
+which are rotated against each other by 90° (shift index is 1) or the faces
+```
+1---2 2---1
+| A | | B |
+4---3 3---4
+```
+which are flipped against each other. Any combination of these can happen. 
+The combination to map neighbor facet to current facet is encoded with
+this data structure via ``rotate \\circ flip`` where the rotation is indiced by
+the shift index.
+"""
+function InterfaceOrientationInfo(grid::AbstractGrid, this_face::FaceIndex, neighbor_face::FaceIndex)
+    face_nodes = faces(getcells(grid)[this_face[1]])[this_face[2]]
+    length(face_nodes) == 1 && return InterfaceOrientationInfo(false, 0)
+    neighbor_face_nodes = faces(getcells(grid)[neighbor_face[1]])[neighbor_face[2]]
+    !all([i ∈ face_nodes for i in neighbor_face_nodes]) && error("Passed faces do not use the same nodes")
+    first_node_idx = findfirst(i -> i == face_nodes[1],neighbor_face_nodes)
+    nfacenodes = length(neighbor_face_nodes)
+    flipped = !(face_nodes[2] == neighbor_face_nodes[first_node_idx < nfacenodes ? first_node_idx + 1 : 1])
+    shift_index = first_node_idx - 1
+    flipped && (shift_index = nfacenodes - shift_index - 1)
+    return InterfaceOrientationInfo(flipped, shift_index)
+end
```

Comment:
This isn't used for now.

Reply:
Maybe this is also a good oppurtunity to merge the different orientation interfaces.

Reply:
Note that this method can fail (I literally debugged a similar an issue yesterday).

For each face you can find 2 affine transformations (fully in reference space). Element 1 -> Face and Face -> Element 2 (as well as the "inverse" mappings). This way the quadrature point on the face of element 1 can be transferred to element 2 by applying both transformations. If noone made touchy-touchy to the face and node definitions recently then you can derive the 3D version from this code https://github.com/Ferrite-FEM/FerriteViz.jl/blob/c5b812eb31786525dd5d2ef3779f39fd9550492a/src/utils.jl#L552-L574 , which is exactly the transformation from face to element I talked about.

Reply:
You're talking about matching coordinates, right?
I'm using the 3D coordinates in 
https://github.com/AbdAlazezAhmed/Ferrite.jl/blob/c28f0b2ebc213caf2c8b7a30845ccdb11cc7ffd7/src/FEValues/interface_values.jl#L149-L156
to get the quadrature point index and use it if we're evaluating using the neighbor facevalues

Reply:
Yes. And if you go from element (e.g. 3D) to the face (e.g. 2D) and again to the element (e.g. 3D again), then you still have 3D coordinates.